FragmentManager is already executing transactions

```
Caused by: java.lang.IllegalStateException: FragmentManager is already executing transactions
                                                                                     at android.app.FragmentManagerImpl.ensureExecReady(FragmentManager.java:1981)
                                                                                     at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2037)
                                                                                     at android.app.FragmentManagerImpl.executePendingTransactions(FragmentManager.java:799)
                                                                                     at com.tbruyelle.rxpermissions2.RxPermissions.getRxPermissionsFragment(RxPermissions.java:54)
                                                                                     at com.tbruyelle.rxpermissions2.RxPermissions.<init>(RxPermissions.java:41)
                                                                                     at com......MessagesActivity.initViews(MessagesActivity.java:113)
```

in this code : 

`    rxPermissions = new RxPermissions(this);
`

I don't know exactly the reason behind this as it was working before without any issue.

--- a/README.md
+++ b/README.md
@@ -17,7 +17,7 @@ allprojects {
 }
 
 dependencies {
-    implementation 'com.github.tbruyelle:rxpermissions:0.10.1'
+    implementation 'com.github.tbruyelle:rxpermissions:0.10.2'
 }
 ```
 
@@ -26,8 +26,10 @@ dependencies {
 Create a `RxPermissions` instance :
 
 ```java
-RxPermissions rxPermissions = new RxPermissions(this); // where this is an Activity instance
+final RxPermissions rxPermissions = new RxPermissions(this); // where this is an Activity or Fragment instance
 ```
+
+**NOTE:** `new RxPermissions(this)` the `this` parameter can be an Activity or a Fragment. If you are using `RxPermissions` inside of a fragment you should pass the fragment instance(`new RxPermissions(this)`) as constructor parameter rather than `new RxPermissions(fragment.getActivity())` or you could face a `java.lang.IllegalStateException: FragmentManager is already executing transactions`.  
 
 Example : request the CAMERA permission (with Retrolambda for brevity, but not required)
 

--- a/lib/src/main/java/com/tbruyelle/rxpermissions2/RxPermissions.java
+++ b/lib/src/main/java/com/tbruyelle/rxpermissions2/RxPermissions.java
@@ -19,6 +19,7 @@ import android.app.Activity;
 import android.os.Build;
 import android.support.annotation.NonNull;
 import android.support.annotation.VisibleForTesting;
+import android.support.v4.app.Fragment;
 import android.support.v4.app.FragmentActivity;
 import android.support.v4.app.FragmentManager;
 import android.text.TextUtils;
@@ -41,11 +42,15 @@ public class RxPermissions {
     Lazy<RxPermissionsFragment> mRxPermissionsFragment;
 
     public RxPermissions(@NonNull final FragmentActivity activity) {
-        mRxPermissionsFragment = getLazySingleton(activity);
+        mRxPermissionsFragment = getLazySingleton(activity.getSupportFragmentManager());
+    }
+
+    public RxPermissions(@NonNull final Fragment fragment) {
+        mRxPermissionsFragment = getLazySingleton(fragment.getChildFragmentManager());
     }
 
     @NonNull
-    private Lazy<RxPermissionsFragment> getLazySingleton(@NonNull final FragmentActivity activity) {
+    private Lazy<RxPermissionsFragment> getLazySingleton(@NonNull final FragmentManager fragmentManager) {
         return new Lazy<RxPermissionsFragment>() {
 
             private RxPermissionsFragment rxPermissionsFragment;
@@ -53,7 +58,7 @@ public class RxPermissions {
             @Override
             public synchronized RxPermissionsFragment get() {
                 if (rxPermissionsFragment == null) {
-                    rxPermissionsFragment = getRxPermissionsFragment(activity);
+                    rxPermissionsFragment = getRxPermissionsFragment(fragmentManager);
                 }
                 return rxPermissionsFragment;
             }
@@ -61,12 +66,11 @@ public class RxPermissions {
         };
     }
 
-    private RxPermissionsFragment getRxPermissionsFragment(@NonNull FragmentActivity activity) {
-        RxPermissionsFragment rxPermissionsFragment = findRxPermissionsFragment(activity);
+    private RxPermissionsFragment getRxPermissionsFragment(@NonNull final FragmentManager fragmentManager) {
+        RxPermissionsFragment rxPermissionsFragment = findRxPermissionsFragment(fragmentManager);
         boolean isNewInstance = rxPermissionsFragment == null;
         if (isNewInstance) {
             rxPermissionsFragment = new RxPermissionsFragment();
-            FragmentManager fragmentManager = activity.getSupportFragmentManager();
             fragmentManager
                     .beginTransaction()
                     .add(rxPermissionsFragment, TAG)
@@ -75,8 +79,8 @@ public class RxPermissions {
         return rxPermissionsFragment;
     }
 
-    private RxPermissionsFragment findRxPermissionsFragment(@NonNull FragmentActivity activity) {
-        return (RxPermissionsFragment) activity.getSupportFragmentManager().findFragmentByTag(TAG);
+    private RxPermissionsFragment findRxPermissionsFragment(@NonNull final FragmentManager fragmentManager) {
+        return (RxPermissionsFragment) fragmentManager.findFragmentByTag(TAG);
     }
 
     public void setLogging(boolean logging) {

